asana-oauth.php

```diff
@@ -122,11 +123,31 @@ private function askAsana($action, $token)
                 echo '<br>cURL error: ' . curl_error($curl);
             }
             echo 'Error on cURL';
-            $response = null;
+            $this->response = null;
         }
 
         curl_close($curl);
 
-        return $response;
+        return $this->response;
+    }
+
+    /**
+     * Checks for errors in the response.
+     * 
+     * @return boolean
+     */
+    public function hasError() {
+        return $this->responseCode != 200 || is_null($this->response);
+    }
+
+    /**
+     * Decodes the response and returns as an object.
+     * 
+     * @return object or null
+     */
+    public function getData() {
+        if(!$this->hasError()){
+            return json_decode($this->response);
+        }
```

Comment:
Missing `return null;` if the previous return is not executed because there is an error.